### PR TITLE
update Az.Compute REST API reference 

### PR DESCRIPTION
**Pull request by grizzlytheodore**
## Mandatory Checklist

- Please choose the target release of Azure PowerShell
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] N/A

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Compute/Compute.Management.Sdk/Generated/ISshPublicKeysOperations.cs [hunk 1]
```diff
@@ -198,7 +203,7 @@ public partial interface ISshPublicKeysOperations
         /// <exception cref="Microsoft.Rest.ValidationException">
         /// Thrown when a required parameter is null
         /// </exception>
-        Task<AzureOperationResponse<SshPublicKeyGenerateKeyPairResult>> GenerateKeyPairWithHttpMessagesAsync(string resourceGroupName, string sshPublicKeyName, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
+        Task<AzureOperationResponse<SshPublicKeyGenerateKeyPairResult>> GenerateKeyPairWithHttpMessagesAsync(string resourceGroupName, string sshPublicKeyName, string encryptionType = default(string), Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
```
Comment (grizzlytheodore) [inline src/Compute/Compute.Management.Sdk/Generated/ISshPublicKeysOperations.cs:206]: need customization

File: src/Compute/Compute.Management.Sdk/Generated/CapacityReservationGroupsOperations.cs [hunk 2]
```diff
@@ -316,7 +317,7 @@ public async Task<AzureOperationResponse<CapacityReservationGroup>> CreateOrUpda
         /// <return>
         /// A response object containing the response body and response headers.
         /// </return>
-        public async Task<AzureOperationResponse<CapacityReservationGroup>> UpdateWithHttpMessagesAsync(string resourceGroupName, string capacityReservationGroupName, IDictionary<string, string> tags = default(IDictionary<string, string>), Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
+        public async Task<AzureOperationResponse<CapacityReservationGroup>> UpdateWithHttpMessagesAsync(string resourceGroupName, string capacityReservationGroupName, CapacityReservationGroupUpdate parameters, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
```
Comment (grizzlytheodore) [inline src/Compute/Compute.Management.Sdk/Generated/CapacityReservationGroupsOperations.cs:320]: breaking change. customization needed

File: src/Compute/Compute.Management.Sdk/Generated/CapacityReservationGroupsOperationsExtensions.cs [hunk 3]
```diff
@@ -86,17 +86,17 @@ public static async Task<CapacityReservationGroup> CreateOrUpdateAsync(this ICap
...
-            /// <param name='tags'>
-            /// Resource tags
+            /// <param name='parameters'>
+            /// Parameters supplied to the Update capacity reservation Group operation.
             /// </param>
-            public static CapacityReservationGroup Update(this ICapacityReservationGroupsOperations operations, string resourceGroupName, string capacityReservationGroupName, IDictionary<string, string> tags = default(IDictionary<string, string>))
+            public static CapacityReservationGroup Update(this ICapacityReservationGroupsOperations operations, string resourceGroupName, string capacityReservationGroupName, CapacityReservationGroupUpdate parameters)
             {
-                return operations.UpdateAsync(resourceGroupName, capacityReservationGroupName, tags).GetAwaiter().GetResult();
+                return operations.UpdateAsync(resourceGroupName, capacityReservationGroupName, parameters).GetAwaiter().GetResult();
```
Comment (grizzlytheodore) [inline src/Compute/Compute.Management.Sdk/Generated/CapacityReservationGroupsOperationsExtensions.cs:94]: breaking change. customization needed

File: src/Compute/Compute.Management.Sdk/Generated/CapacityReservationGroupsOperationsExtensions.cs [hunk 4]
```diff
@@ -107,15 +107,15 @@ public static CapacityReservationGroup Update(this ICapacityReservationGroupsOpe
...
-            /// <param name='tags'>
-            /// Resource tags
+            /// <param name='parameters'>
+            /// Parameters supplied to the Update capacity reservation Group operation.
             /// </param>
             /// <param name='cancellationToken'>
             /// The cancellation token.
             /// </param>
-            public static async Task<CapacityReservationGroup> UpdateAsync(this ICapacityReservationGroupsOperations operations, string resourceGroupName, string capacityReservationGroupName, IDictionary<string, string> tags = default(IDictionary<string, string>), CancellationToken cancellationToken = default(CancellationToken))
+            public static async Task<CapacityReservationGroup> UpdateAsync(this ICapacityReservationGroupsOperations operations, string resourceGroupName, string capacityReservationGroupName, CapacityReservationGroupUpdate parameters, CancellationToken cancellationToken = default(CancellationToken))
```
Comment (grizzlytheodore) [inline src/Compute/Compute.Management.Sdk/Generated/CapacityReservationGroupsOperationsExtensions.cs:116]: breaking change. customization needed

File: src/Compute/Compute.Management.Sdk/Generated/ICapacityReservationGroupsOperations.cs [hunk 5]
```diff
@@ -82,7 +84,7 @@ public partial interface ICapacityReservationGroupsOperations
         /// <exception cref="Microsoft.Rest.ValidationException">
         /// Thrown when a required parameter is null
         /// </exception>
-        Task<AzureOperationResponse<CapacityReservationGroup>> UpdateWithHttpMessagesAsync(string resourceGroupName, string capacityReservationGroupName, IDictionary<string, string> tags = default(IDictionary<string, string>), Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
+        Task<AzureOperationResponse<CapacityReservationGroup>> UpdateWithHttpMessagesAsync(string resourceGroupName, string capacityReservationGroupName, CapacityReservationGroupUpdate parameters, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
```
Comment (grizzlytheodore) [inline src/Compute/Compute.Management.Sdk/Generated/ICapacityReservationGroupsOperations.cs:87]: breaking change. customization needed

File: src/Compute/Compute.Management.Sdk/Generated/IVirtualMachineScaleSetVMsOperations.cs [hunk 6]
```diff
@@ -606,7 +711,7 @@ public partial interface IVirtualMachineScaleSetVMsOperations
         /// <exception cref="Microsoft.Rest.ValidationException">
         /// Thrown when a required parameter is null
         /// </exception>
-        Task<AzureOperationResponse<VirtualMachineScaleSetVM>> BeginUpdateWithHttpMessagesAsync(string resourceGroupName, string vmScaleSetName, string instanceId, VirtualMachineScaleSetVM parameters, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
+        Task<AzureOperationResponse<VirtualMachineScaleSetVM>> BeginUpdateWithHttpMessagesAsync(string resourceGroupName, string vmScaleSetName, string instanceId, VirtualMachineScaleSetVM parameters, string ifMatch = default(string), string ifNoneMatch = default(string), Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
```
Comment (grizzlytheodore) [inline src/Compute/Compute.Management.Sdk/Generated/IVirtualMachineScaleSetVMsOperations.cs:714]: breaking change. customization needed

**Comment by dolauli** [comment]
@grizzlytheodore Please follow the [guide](https://eng.ms/docs/cloud-ai-platform/azure-core/azure-management-and-platforms/control-plane-bburns/azure-cli-tools-azure-cli-powershell-and-terraform/azure-cli-tools/devguide/azps/coding/generate-sdk-with-autorest-powershell) to replace the track 1 SDK with the SDK generated by autorest.powershell.

**Comment by grizzlytheodore** [comment]
@dolauli I went through the process of migration, and I have not finished resolving everything yet, but we have 
700 changed files in generated SDK 
100 files for az.compute for change property names 
using the .\function-compare.ps1, 60 changed functions. 

I would assume about 1000 changed files for this migration, which I am hesitant to make right before the holidays. I think it is better if we go through this autorest.csharp -> autorest.powershell migration in the first quarter of next year instead 

Please let me know your thoughts  

@bilaakpan-ms  @dcaro

**Review by dolauli — APPROVED** [review]

**Pull request merged**